### PR TITLE
Fix inh and format some more

### PR DESCRIPTION
**Pull request by roberth**

### CODE REVIEW

File: canonix/src/Format.hs [hunk 1]
```diff
@@ -108,15 +131,54 @@ verbatim n = do
   src <- asksParent source
   let bs = BS.drop (startByte n) (BS.take (endByte n) src)
   write bs
+  _ <- whenSingleLineAllowed $ do
+    rw <- asksParent remainingWidth
+    tellParent mempty { singleLine = bs <$ guard (BS.length bs <= rw) }
   pure ()
 
+whenSingleLineAllowed :: CnxFmt a -> CnxFmt (Maybe a)
```
Comment (domenkozar) [inline canonix/src/Format.hs:130]: Would be nice to have some comments for these combinations to know when to use them, it's not entirely clear to me.

Comment (roberth) [reply canonix/src/Format.hs:130]: I agree it needs more documentation. We should get this thing out of the proof-of-concept sphere.

This particular function just disables some behavior when a node must be formatted in its multi-line style. It's a helper for constructing the real output functions that you actually want to use.
Needs more modules.

Comment (domenkozar) [reply canonix/src/Format.hs:130]: Are these combinators expected to change soon? If not adding documentation (just a single line of what it does) sounds like a good time to do it.

Comment (roberth) [reply canonix/src/Format.hs:130]: They'll change soon (I hope) because of #19 #18. I'll add some inline docs.

**Review by domenkozar — APPROVED** [review]

**Pull request merged**